mozetl/search/daily_search_rollups.py

```diff
@@ -1,309 +1,371 @@
-        frame = add_profile_day(frame)
-    frame.repartition("profile_day")
-    nulls_frame = frame.where('search_counts is null')
-    temp_table = TEMP_TABLE_TEMPLATE.format(int(time.time()))
-    select_rollup = SELECT_ROLLUP_TEMPLATE.format(temp_table)
+    :main_summary DataFrame: source table
+    :mode str: `daily` or `monthly`
+    """
 
-    exploded = frame.selectExpr(
-        "submission_date",
-        "profile_day",
+    # take a subset of the original dataframe
+    columns = [
+        F.col("client_id").alias("profile"),
         "country",
         "locale",
         "distribution_id",
-        "default_search_engine as default_provider",
-        "explode(search_counts) as search_counts")
-    unwrapped = exploded.selectExpr(
-        "submission_date",
-        "profile_day",
+        F.col("default_search_engine").alias("default_provider"),
+        "search_counts",
+    ]
+
+    # attributes of the final search rollup
+    attributes = [
         "country",
+        "search_provider",
+        "default_provider",
         "locale",
         "distribution_id",
-        "default_provider",
-        "search_counts.engine as search_provider",
-        "search_counts.source as search_source",
-        "search_counts.count as search_count")
-    unwrapped_nulls = nulls_frame.selectExpr(
-        "submission_date",
-        "profile_day",
+    ]
+
+    defaults = {
+        "country": "XX",
+        "search_provider": "NO_SEARCHES",
+        "default_provider": "NO_DEFAULT",
+        "locale": "xx",
+        "distribution_id": "MOZILLA",
+        "search_count": 0,
+    }
+
+    dataframe = main_summary.select(columns)
+    exploded = explode_searches(dataframe).na.fill(defaults)
+
+    # don't include shares if we process the monthly rollup
+    if mode == "monthly":
+        processed = exploded
+    else:
+        processed = with_shares(exploded)
+    search_rollup = rollup_searches(processed, attributes, mode)
+
+    return search_rollup
+
+
+def format_spark_path(bucket, prefix):
+    return "s3://{}/{}".format(bucket, prefix)
+
+
+def get_end_date(ds_start, period):
+    """ Return the end date given the start date and period. """
+    date_start = arrow.get(ds_start, "YYYYMMDD")
+    if period == "monthly":
+        date_end = date_start.replace(months=+1)
+    else:
+        date_end = date_start.replace(days=+1)
+    ds_end = date_end.format("YYYYMMDD")
+
+    return ds_end
+
+
+def extract(spark, path, ds_start, period):
+    """Extract the source dataframe from the spark compatible path.
+
+    spark: SparkSession
+    path: path to parquet files in s3
+    ds_start: inclusive date
+    """
+    ds_end = get_end_date(ds_start, period)
+
+    return (
+        spark.read
+        .option("mergeSchema", "true")
+        .parquet(path)
+        .where((F.col("submission_date_s3") >= ds_start) &
+               (F.col("submission_date_s3") < ds_end))
+    )
+
+
+def get_last_manifest_version(bucket, prefix, pattern):
```

Comment:
I don't really want to needlessly break scripts, the handoff piece seems to be out of the scope of this bug anyways. I'm happy to leave this as it is.